contract/src/main/java/io/neow3j/contract/NeoNameService.java

```diff
@@ -0,0 +1,368 @@
+package io.neow3j.contract;
+
+import static io.neow3j.contract.ContractParameter.byteArray;
+import static io.neow3j.contract.ContractParameter.hash160;
+import static io.neow3j.contract.ContractParameter.integer;
+import static io.neow3j.contract.ContractParameter.string;
+import static io.neow3j.model.types.StackItemType.MAP;
+import static java.util.Collections.singletonList;
+
+import io.neow3j.contract.exceptions.UnexpectedReturnTypeException;
+import io.neow3j.protocol.Neow3j;
+import io.neow3j.protocol.core.RecordType;
+import io.neow3j.protocol.core.methods.response.InvocationResult;
+import io.neow3j.protocol.core.methods.response.MapStackItem;
+import io.neow3j.protocol.core.methods.response.NameState;
+import io.neow3j.protocol.core.methods.response.StackItem;
+import io.neow3j.utils.Numeric;
+import io.neow3j.wallet.Wallet;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+/**
+ * Represents the NameService native contract and provides methods to invoke its functions.
+ */
+public class NeoNameService extends NonFungibleToken {
+
+    public final static String NAME = "NameService";
+    public final static long NEF_CHECKSUM = 3740064217L;
+    public static final ScriptHash SCRIPT_HASH = getScriptHashOfNativeContract(NEF_CHECKSUM, NAME);
+
+    private static final String ADD_ROOT = "addRoot";
+    private static final String SET_PRICE = "setPrice";
+    private static final String GET_PRICE = "getPrice";
+    private static final String IS_AVAILABLE = "isAvailable";
+    private static final String REGISTER = "register";
+    private static final String RENEW = "renew";
+    private static final String SET_ADMIN = "setAdmin";
+    private static final String SET_RECORD = "setRecord";
+    private static final String GET_RECORD = "getRecord";
+    private static final String DELETE_RECORD = "deleteRecord";
+    private static final String RESOLVE = "resolve";
+
+    private static final String PROPERTIES = "properties";
+
+    private static final BigInteger MAXIMAL_PRICE = new BigInteger("1000000000000");
+    private static final Pattern ROOT_REGEX_PATTERN = Pattern.compile("^[a-z][a-z0-9]{0,15}$");
+    public static final Pattern NAME_REGEX_PATTERN = Pattern.compile(
```

Comment:
Any reason why this attribute is `public`? If no, leave it private, or change all `Pattern` attributes to protected (if required on tests).

Reply:
No, I missed changing it back to private. I'll change it.